src/xcode/ENA/ENA/Source/Scenes/Home/HomeInteractor.swift

```diff
@@ -427,3 +431,30 @@ extension HomeInteractor {
 		homeViewController.showExposureNotificationSetting()
 	}
 }
+
+// MARK: - CountdownTimerDelegate methods.
+
+/// The `CountdownTimerDelegate` is used to update the remaining time that is shown on the risk cell button until a manual refresh is allowed.
+extension HomeInteractor: CountdownTimerDelegate {
+	private func scheduleCountdownTimer() {
+		let nextUpdate = riskProvider.nextExposureDetectionDate()
+		countdownTimer = CountdownTimer(countdownTo: nextUpdate)
+		countdownTimer?.delegate = self
+		countdownTimer?.start()
```

Comment:
I see, the timer `invalidate`  is only called in the CoutdownTimer class:
```swift
 deinit {
		timer?.invalidate()
		timer = nil
	} 
```
My question is, if the app goes background, the Timer still runs? What will happened if the user opens the App after 2 days?

Reply:
thanks for bringing this up, I discovered a bug in my implementation where the timer was not stopped as I expected due to a reference cycle. I added a manual invalidate() method to make sure the timer is handled correctly.
Regarding the opening of the app after some time has passed: The timer is guaranteed to be up-to-date since the scene delegate will trigger the updateExposureState() method of the home VC when the app comes into foreground.
To make sure the timer does not drain the users battery when the app goes into background, I added a few lines that invalidate the handler when the app goes into background.